File: mobile/src/main/java/com/siliconlabs/bledemo/MainMenu/Adapters/MenuAdapter.java
package com.siliconlabs.bledemo.MainMenu.Adapters;

import android.content.Context;

import androidx.core.content.ContextCompat;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.siliconlabs.bledemo.MainMenu.Model.MenuItem;
import com.siliconlabs.bledemo.MainMenu.Model.MenuItemType;
import com.siliconlabs.bledemo.R;

import java.util.List;

public class MenuAdapter extends RecyclerView.Adapter<MenuAdapter.MenuItemViewHolder> {
    private List<MenuItem> menuItemList;
    private OnMenuItemClickListener onMenuItemClickListener;
    private Context context;

    public class MenuItemViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
        ImageView menuIconIV;
        TextView menuTitleTV;
        TextView menuDescrTV;

        public MenuItemViewHolder(View itemView) {
            super(itemView);
            menuIconIV = itemView.findViewById(R.id.imageview_menu_icon);
            menuTitleTV = itemView.findViewById(R.id.textview_menu_title);
            menuDescrTV = itemView.findViewById(R.id.textview_menu_description);
            itemView.setOnClickListener(this);
        }

        @Override
        public void onClick(View v) {
            int position = getAdapterPosition();
            onMenuItemClickListener.onMenuItemClick(menuItemList.get(position).getMenuItemType());
        }
    }

    public MenuAdapter(List<MenuItem> menuItemList, OnMenuItemClickListener onMenuItemClickListener, Context context) {
        this.menuItemList = menuItemList;
        this.onMenuItemClickListener = onMenuItemClickListener;
        this.context = context;
    }


    @Override
    public MenuAdapter.MenuItemViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.adapter_main_menu, parent, false);
        return new MenuItemViewHolder(view);
    }


    @Override
    public void onBindViewHolder(MenuItemViewHolder holder, int position) {
        MenuItem menuItem = menuItemList.get(position);
        holder.menuTitleTV.setText(menuItem.getTitle());
        holder.menuDescrTV.setText(menuItem.getDescription());

        holder.menuIconIV.setImageDrawable(ContextCompat.getDrawable(context, menuItem.getMenuIcon()));
    }


    @Override
    public int getItemCount() {
        return menuItemList.size();
    }

    public interface OnMenuItemClickListener {
        void onMenuItemClick(MenuItemType menuItemType);
    }

}

File: mobile/src/main/java/com/siliconlabs/bledemo/main_menu/adapters/MenuAdapter.kt
package com.siliconlabs.bledemo.main_menu.adapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.siliconlabs.bledemo.main_menu.adapters.MenuAdapter.MenuItemViewHolder
import com.siliconlabs.bledemo.main_menu.menu_items.MainMenuItem
import com.siliconlabs.bledemo.R

class MenuAdapter(private val list: List<MainMenuItem>, private val listener: OnMenuItemClickListener, private val context: Context) : RecyclerView.Adapter<MenuItemViewHolder>() {

    inner class MenuItemViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {
        private val  ivMenuIcon = itemView.findViewById(R.id.iv_menu_icon) as ImageView
        private val tvMenuTitle = itemView.findViewById(R.id.tv_menu_title) as TextView
        private val tvMenuDescr = itemView.findViewById(R.id.tv_menu_description) as TextView

        init {
            itemView.setOnClickListener(this)
        }

        fun bind(item: MainMenuItem) {
            tvMenuTitle.text = item.title
            tvMenuDescr.text = item.description
            ivMenuIcon.setImageDrawable(ContextCompat.getDrawable(context, item.imageResId))
        }

        override fun onClick(v: View) {
            listener.onMenuItemClick(list[adapterPosition])
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MenuItemViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.adapter_main_menu, parent, false)
        return MenuItemViewHolder(view)
    }

    override fun onBindViewHolder(holder: MenuItemViewHolder, position: Int) {
        val menuItem = list[position]
        holder.bind(menuItem)
    }

    override fun getItemCount(): Int {
        return list.size
    }

    interface OnMenuItemClickListener {
        fun onMenuItemClick(menuItem: MainMenuItem)
    }

}